common/logging/EventAuditLogger.php

```diff
@@ -0,0 +1,1090 @@
+
+        $category = $event;
+        // Special case delete for lists table
+        if ($event == 'delete') {
+            $category = $this->eventCategoryFinder($comments, $event, '');
+        }
+
+        // deal with comments encryption, if turned on
+        $encrypt_comment = 'No';
+        if (!empty($comments)) {
+            if ($GLOBALS["enable_auditlog_encryption"]) {
+                $comments =  encryptStandard($comments);
+                $encrypt_comment = 'Yes';
+            }
+        }
+
+        if ($log_from == 'patient-portal') {
+            $sqlMenuItems = "SELECT * FROM patient_portal_menu";
+
+            $resMenuItems = sqlStatement($sqlMenuItems);
+            for ($iter=0; $rowMenuItem=sqlFetchArray($resMenuItems); $iter++) {
+                $menuItems[$rowMenuItem['patient_portal_menu_id']] = $rowMenuItem['menu_name'];
+            }
+
+            $menuItemId = array_search($menu_item, $menuItems);
+            $sql = "insert into log ( date, event,category, user, patient_id, groupname, success, comments,
+                log_from, menu_item_id, crt_user, ccda_doc_id) values ( NOW(), ?,'Patient Portal', ?, ?, ?, ?, ?, ?, ?,?, ?)";
+            $ret = sqlStatementNoLog($sql, array($event, $user, $patient_id, $groupname, $success, $comments,$log_from, $menuItemId,$crt_user, $ccda_doc_id));
+        } else {
+            /* More details added to the log */
+            $sql = "insert into log ( date, event,category, user, groupname, success, comments, crt_user, patient_id) " .
+                "values ( NOW(), " . $adodb->qstr($event) . ",". $adodb->qstr($category) . "," . $adodb->qstr($user) .
+                "," . $adodb->qstr($groupname) . "," . $adodb->qstr($success) . "," .
+                $adodb->qstr($comments) ."," .
+                $adodb->qstr($crt_user) ."," . $adodb->qstr($patient_id). ")";
+
+            $ret = sqlInsertClean_audit($sql);
+        }
+
+        // Send item to log_comment_encrypt for comment encyption tracking
+        $last_log_id = $GLOBALS['adodb']['db']->Insert_ID();
+        $encryptLogQry = "INSERT INTO log_comment_encrypt (log_id, encrypt, checksum, `version`) ".
+            " VALUES ( ".
+            $adodb->qstr($last_log_id) . "," .
+            $adodb->qstr($encrypt_comment) . "," .
+            "'', '3')";
+        sqlInsertClean_audit($encryptLogQry);
+
+        if (($patient_id=="NULL") || ($patient_id==null)) {
+            $patient_id=0;
+        }
+
+        $this->send_atna_audit_msg($user, $groupname, $event, $patient_id, $success, $comments);
+    }
+
+    /**
+     * @deprecated - doesn't appear to be in use.
+     * @param      $date
+     * @param      string $user
+     * @param      string $cols
+     * @return     mixed
+     */
+    public function getEventByDate($date, $user = "", $cols = "DISTINCT date, event, user, groupname, patient_id, success, comments, checksum")
```

Comment:
@sjpadgett , do you think it makes sense to remove functions here (like this one) that are not being used in codebase (ie. really clean things up so can focus on streamlining performance)?

Reply:
did a quick check, this is the only function that is not being used in this class. I vote to remove it :)